packages/server/src/index.ts

```diff
@@ -62,9 +58,9 @@ export interface ServerAdapterObject<
   ): Promise<Response> | Response;
 }
 
-export type ServerAdapter<TServerContext, TBaseObject extends ServerAdapterBaseObject<TServerContext>> = TBaseObject &
```

Comment:
So returned server adapter object no longer have other methods of given object;
```ts
const app = createServerAdapter(Router());
app.get('/', req => ...);
```
```

Reply:
Ah, so that is the reason why we need the TBaseObject generic.

The thing is, with the current setup (on `master`), inference does not work with the context. Try this:
```ts
interface Ctx {
  some: string;
}

const handler = (_req: Request, _ctx: Ctx) => new Response();
const adapter = createServerAdapter(handler); // TS error with handler

const request = new Request('http://localhost');
await adapter(request, ctx); // TS error with request
```

even passing in the `Ctx` as generic creates problems:
```diff
- const adapter = createServerAdapter(handler);
+ const adapter = createServerAdapter<Ctx>(handler);
```